aten/src/ATen/TensorNames.cpp

```diff
@@ -62,22 +62,28 @@ TensorNames::TensorNames(ArrayRef<Dimname> names, int64_t start, int64_t end) {
   }
 }
 
-TensorNames TensorNames::unifyFromRight(const TensorNames& other, const char* op_name) const {
-  const auto longer_size = std::max(names_.size(), other.names_.size());
-  TensorNameVec result;
-  result.reserve(longer_size);
-
-  const auto& longer = names_.size() == longer_size ? names_ : other.names_;
-  const auto& shorter = names_.size() == longer_size ? other.names_ : names_;
-  const auto size_difference = longer_size - shorter.size();
-
-  result.insert(result.begin(), longer.begin(), longer.begin() + size_difference);
-
-  for (int64_t idx = size_difference; idx < longer_size; ++idx) {
-    result.push_back(longer[idx].unify(shorter[idx - size_difference], op_name));
+TensorNames& TensorNames::unifyFromRightInplace(const TensorNames& other, const char* op_name) {
+  int64_t longer_size = std::max(names_.size(), other.names_.size());
+  int64_t shorter_size = std::min(names_.size(), other.names_.size());
+  int64_t size_difference = longer_size - shorter_size;
+  bool this_is_longer = names_.size() == longer_size;
+
+  if (this_is_longer) {
```

Comment:
I might prefer something like `size_diff = std::abs(names.size() - other.names.size());` and `if (names.size() > other.names.size())` if that also works, instead of tracking min/max sizes and a bool.

Reply:
Sure, that sounds reasonable